corehq/form_processor/submission_post.py

```diff
@@ -541,3 +541,20 @@ def unfinished_submission(instance):
     tracker = SubmissionProcessTracker(unfinished_submission_stub)
     yield tracker
     tracker.submission_fully_processed()
+
+
+@contextlib.contextmanager
+def unfinished_archive(instance, user_id, archive):
+    unfinished_archive_stub = None
+    if not getattr(instance, 'deprecated_form_id', None):
+        unfinished_archive_stub = UnfinishedArchiveStub.objects.create(
+            user_id=user_id,
+            xform_id=instance.form_id,
+            timestamp=datetime.datetime.utcnow(),
+            # if archive is False, this is an unarchive stub.
+            archive=archive,
+            domain=instance.domain,
+        )
+    tracker = SubmissionProcessTracker(unfinished_archive_stub)
```

Comment:
I don't think the use of the `tracker` is necessary here. You can just `yield` and then delete the stub afterwards.

Reply:
I mean replace it with a plain yield and a delete:
```
yield 
unfinished_archive_stub.delete()
```

Reply:
https://github.com/dimagi/commcare-hq/pull/22470/commits/e5c7e9fd66f61ee296eb2f722622cdcf5a7910b1